openmicroscopy/openmicroscopy
# Add DN for groups (fix #6719).

This PR adds an "ldap" filed to the `ExperimenterGroup` model entity. This also means that the respective DB table will be extended with such column. This change allows to control which groups came from LDAP on user login and allows to change their LDAP "state". Currently, the only way to do it is using the OMERO LDAP CLI.

To test the code changes:
- verify that all tests are green (travis, integration tests),
- try logging in with and LDAP user's credentials,
- try creating (`bin/omero ldap create`) an LDAP user and logging in as that user,
- try creating a normal (non-LDAP) user via the CLI and clients and verify that no errors occur (e.g. Hibernate errors related to missing value of the ldap field, either for group or experimenter),
- try using `bin/omero ldap setdn --group-name <name> false` to disable the LDAP field on a group and verify that the group appears in the output of `bin/omero ldap discover`,
- try using `bin/omero ldap setdn` and `bin/omero ldap getdn` with all the CLI switches (group name, group ID etc.),

To test the DB upgrade script:
- try using a server without this PR, create LDAP users and move to the code with this PR included - run the DB upgrade script (OMERO5.1DEV__13.sql) and verify that all users still exist and you can log in as them,
- the same test could be repeated with the 5.0.5 server as the starting point and then moving to 5.1DEV__14.
